R/get.R

```diff
@@ -334,13 +328,16 @@ content_list_with_permissions <- function(src, ..., .p = NULL) {
   content_list <- get_content(src, .p = .p)
 
   message("Getting permission list")
-  pb <- progress::progress_bar$new(
+  pb <- optional_progress_bar(
     total = nrow(content_list),
     format = "[:bar] :percent :eta"
   )
-  content_list[["permission"]] <- purrr::map(
-    content_list$guid,
-    function(.x) .get_content_permission_with_progress(src, .x, pb)
+  content_list[["permission"]] <- purrr::pmap(
```

Comment:
@jonkeane I refactored this to remove the need for the `finished` field; I also switched this to a `pmap` in order to save a `GET /v1/content/guid/` request for each content item--we already have the data so we don't need to fetch it again. There are surely lots more of these in the code, I'll do some hunting (later).

Reply:
Nice!

> There are surely lots more of these in the code, I'll do some hunting (later).

I did search for `.get_content_permission_with_progress` and I thought I only saw it here(!!). But there definitely might be other things that pass the progress bar like this

Reply:
Not `.get_content_permission_with_progress`, I mean the pattern of doing `get_content()` to get a data frame of all content, then iterating over `guid` and calling `content_item(connect, guid)`, which does a `GET` to fetch the Content record. But we already have that, every row in that data.frame is Content.

Reply:
AAAAH, right